### PR TITLE
Show health stat on wakeup and SELFAWARE

### PR DESCRIPTION
**Pull request by Coolthulhu**
Health stat is not described much and is a confusing mechanic. The only good way of seeing its effects right now is by timing regeneration during sleep. For this reason, it hasn't been balanced much and not even doing hard drugs will change it significantly (but regular vitamins will).

Here I added a message on wakeup when well rested that described current health state. This message shows the exact number if the survivor has SELFAWARE trait, otherwise it isn't very precise.

Changed the SELFAWARE trait to active. Activating it will print the same message as on well rested wakeup.

The messages are printed for the following ranges:
- >50 - mentions sped up regeneration
- -10 - 49 - no message here
- -30 - -11 - mentions bad diet
- <-30 - mentions slowed down regeneration and hints at leaky bionic/sickly artifact

Also added an option to view/change health and health mod in debug menu.

Now, the possible problems:
- The messages seem a bit awkward to me. Any better ideas?
- Currently only waking up well rested will print the message. Any other good places for it? Other wakeups? Stethoscope usage (it's a bit rare)? Blood analysis?

### CODE REVIEW

File: src/player.cpp [hunk 1]
```diff
@@ -7200,53 +7222,10 @@ void player::hardcoded_effects(effect &it)
         // a little, and came out of it well into Parched.  Hibernating shouldn't endanger your
         // life like that--but since there's much less fluid reserve than food reserve,
         // simply using the same numbers won't work.
-        if((int(calendar::turn) % 350 == 0) && has_trait("HIBERNATE") && !(hunger > -60) && !(thirst >= 80)) {
```
Comment (BevapDin) [inline src/player.cpp]: Why did you remove this block?

Comment (Coolthulhu) [inline src/player.cpp]: Oh, forgot to mention it in the description.

I integrated it into the other one, because they were nearly identical.

**Comment by Zireael07** [comment]
Great to see this, I never understood how health stat worked.

File: src/player.cpp [hunk 2]
```diff
@@ -5828,6 +5828,28 @@ static int bound_mod_to_vals(int val, int mod, int max, int min)
...
 }
 
+void player::print_health()
+{
+    if( !is_player() ) {
+        return;
+    }
+
+    if( healthy > 50 ) {
+        add_msg( m_good, _("You're very healthy, your wounds heal faster") );
```
Comment (Rivet-the-Zombie) [inline src/player.cpp]: Shouldn't these lines have closing punctuation like a period or something at the end?

Comment (kevingranade) [inline src/player.cpp]: The messages are fine for the selfaware thing, but they could use some
personality for the wake-up message.
"You feel fantastic!  All that healthy living is doing you good."
"Your wounds are healing pretty fast, you must be taking good care of
yourself."
"You feel a little cruddy, getting out of bed is a struggle."
"You don't feel so good, not enough to slow you down, but at this rate
you're headed for a cold."
"You feel terrible, maybe you should think about eating better."
"Your wounds aren't healing as fast as they should, maybe some vitamins are
in order."
Ideally they'd be randomized a bit and situational, displaying the messages
about healing only when you aren't at full HP, but that's most of the time
I'd think.

**Comment by kevingranade** [comment]
I'm tempted to say stash a timestamp of when you went to sleep and check for it being far enough in the past to not be spammy instead of using well-rested, I'm not sure it's representative, but my characters rarely hit well rested before being woken up by something..

**Comment by DavidKeaton** [comment]
Same here Kevin, well rested is a rare occasion for me. Albeit I am a nomad...

**Comment by Coolthulhu** [comment]
OK, changed the trigger to waking up after 2 hours or more rather than waking up well rested.

Didn't change the messages. While more messages could be useful, I think they should be rather "boring" - to reflect the survivor being kinda sluggish on wakeup.

**Comment by DavidKeaton** [comment]
Oh I like that idea!

**Comment by DavidKeaton** [comment]
"You realize that all your parts seem where they should be. You stretch and get ready."

**Comment by i2amroy** [comment]
The messages still need closing punctuation.

And I'm fine with more "boring" messages on wakeup, but they still could use a *little* bit more personality. Right now some of them are downright clinical.

Other than that though this looks good for merging. If you'd prefer I can fix the closing punctuation while merging and then cook up a quick PR for some more and more flavorful messages, I'd rather not hold this up any longer for flavor reasons.

**Comment by Coolthulhu** [comment]
Forgot about the punctuation. If it's not a problem, please fix.

I'm not too creative with messages, if you have better ideas, then go for it. I will probably be doing a bulk mergetest after I manage (or drop out of frustration) my current bashing PR (to fix the un-collapsing roofs).

**Comment by i2amroy** [comment]
Sure thing I'll test this one and merge it (assuming no other problems pop up) right now.

**Comment by i2amroy** [comment]
Okay, not getting any messages at all, with or without the SELFAWARE trait, something is going strange here.

**Comment by kevingranade** [comment]
???

**Comment by karlnp** [comment]
I love your comments, Kevin.
